presto-hive/src/main/java/com/facebook/presto/hive/s3select/S3SelectRecordCursorProvider.java

```diff
@@ -97,4 +110,65 @@ public Optional<RecordCursor> createRecordCursor(
         // unsupported serdes
         return Optional.empty();
     }
+
+    private static boolean hasFilters(
+            Properties schema,
+            TupleDomain<HiveColumnHandle> effectivePredicate,
+            List<HiveColumnHandle> projectedColumns)
+    {
+        // When there are no effective predicates and the projected columns and column types are identical to the schema,
+        // it means we get all the data out of S3. We can use S3 GetObject instead of S3 SelectObjectContent in these cases.
+        if (effectivePredicate.isAll()) {
+            return !isSchemaEquivalent(projectedColumns, schema);
+        }
+        return true;
+    }
+
+    private static boolean isSchemaEquivalent(List<HiveColumnHandle> columns, Properties schema)
+    {
+        Set<String> projectedColumnNames = getColumnProperty(columns, HiveColumnHandle::getName);
+        Set<String> projectedColumnTypes = getColumnProperty(columns, column -> column.getHiveType().getTypeInfo().getTypeName());
```

Comment:
Would we still want to have S3 select enabled if all columns are selected but the schema types don't align? (would this be a "cast" situation?) It seems like regardless of the types involved, selecting all columns won't reduce data over the wire.